v2/pkg/engine/resolve/arena_test.go

```diff
@@ -0,0 +1,261 @@
+package resolve
+
+import (
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/wundergraph/go-arena"
+)
+
+func TestNewArenaPool(t *testing.T) {
+	pool := NewArenaPool()
+
+	require.NotNil(t, pool, "NewArenaPool returned nil")
+	assert.Equal(t, 0, len(pool.pool), "expected empty pool")
+	assert.Equal(t, 0, len(pool.sizes), "expected empty sizes map")
```

Comment:
Nit: `assert.Len()` is recommended for this case by [testifylint](https://github.com/Antonboom/testifylint?tab=readme-ov-file#len).

Reply:
changed